app/services/status_update_service.rb

```diff
@@ -33,12 +35,16 @@ def call
 
   attr_reader :reference, :status, :user
 
-  def delivery_attempt
-    @delivery_attempt ||= DeliveryAttempt.find_by!(reference: reference)
+  def permanent_failure?
+    status == "permanent-failure"
+  end
+
+  def temporary_failure?
+    status == "temporary-failure"
   end
 
   def email
-    @email ||= delivery_attempt.email
+    @email ||= DeliveryAttempt.find_by!(reference: reference).email
```

Comment:
This seems to unnecessarily initialise delivery attempt. Can we do it more like `Email.joins(:delivery_attempts).find_by("delivery_attemps.reference": reference)`?

Reply:
Yep, that's a lot better. https://github.com/alphagov/email-alert-api/pull/424